packages/eslint-plugin/src/rules/explicit-function-return-type.ts

```diff
@@ -60,11 +67,61 @@ export default util.createRule<Options, MessageIds>({
       allowHigherOrderFunctions: true,
       allowDirectConstAssertionInArrowFunctions: true,
       allowConciseArrowFunctionExpressionsStartingWithVoid: false,
+      allowedNames: [],
     },
   ],
   create(context, [options]) {
     const sourceCode = context.getSourceCode();
+    function isAllowedName(
+      node:
+        | TSESTree.ArrowFunctionExpression
+        | TSESTree.FunctionExpression
+        | TSESTree.FunctionDeclaration,
+    ): boolean {
+      if (!options.allowedNames || !options.allowedNames.length) {
+        return false;
+      }
 
+      if (
+        node.type === AST_NODE_TYPES.ArrowFunctionExpression ||
+        node.type === AST_NODE_TYPES.FunctionExpression
+      ) {
+        const parent = node.parent;
+        let funcName;
+        if (node.id?.name) {
+          funcName = node.id.name;
+        } else if (parent) {
+          switch (parent.type) {
+            case AST_NODE_TYPES.VariableDeclarator: {
+              if (parent.id.type === AST_NODE_TYPES.Identifier) {
+                funcName = parent.id.name;
+              }
+              break;
+            }
+            case AST_NODE_TYPES.MethodDefinition:
+            case AST_NODE_TYPES.PropertyDefinition:
+            case AST_NODE_TYPES.Property: {
+              if (parent.key.type === AST_NODE_TYPES.Identifier) {
```

Comment:
one minor problem with this code:

```ts
const ignoredName = 'notIgnoredName';
class Foo {
  [ignoredName]() {}
}
```

This is valid code! Which obviously should *not* be ignored.
Only checking for `.key.type === 'Identifier'` will match this case.
You should also assert that `.computed === false`.

cc @JoshuaKGoldberg as an FYI - this is a really common case that everyone (including me!) always forgets for properties of all kinds AND for member expressions as well.
I raised the issue with the estree spec but the outcome there was really "spec can't change".

Reply:
Ah thanks for the spot!

Reply:
Thank you!
I didn't know such a pattern existed.
@bradzacher 
I changed the computed name so that it is not ignored by the allowedName.